bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TypedescValueImpl.java

```diff
@@ -91,6 +93,9 @@ public Object instantiate(Strand s, BInitialValueEntry[] initialValues) {
             return new MapValueImpl(this.describingType, (BMapInitialValueEntry[]) initialValues);
         } else if (referredType.getTag() == TypeTags.TUPLE_TAG) {
             return new TupleValueImpl(this.describingType, (BListInitialValueEntry[]) initialValues);
+        } else if (getEffectiveType(referredType).getTag() == TypeTags.TUPLE_TAG) {
+            return new TupleValueImpl(new BTypeReferenceType(getEffectiveType(referredType)),
```

Comment:
Why do we create a new `TypeReferenceType` and pass it? 
IIRC, as the `getEffectiveType(referredType)` is expected to be a TupleType here, we can pass the `TupleType` itself.

Reply:
```ballerina
type A [int, string|xml] & readonly;
public function main() {
    A tupleValue9 = [1, ""];

}
```
Previously, we did not generate a type reference for the intersection of a tuple with a readonly type. With these changes, for the above example, I have generated a type reference for type 'A'. Due to this, the referred type of type reference  is an intersection type, so I have created a new type reference with a tuple as the referred type

Reply:
Is it correct to generate two type descriptors for the following example?
```ballerina
type A [int, string|xml] & readonly;
public function main() {
    A tupleValue1 = [1, ""];
    A tupleValue2 = [1, ""];
}
```

Reply:
OK, if it for `A`, we'll need the type reference type. I misunderstood it as for the effective type (that we don't need to be the same).

Reply:
> Is it correct to generate two typedescs for the following example?
> 
> ```ballerina
> type A [int, string|xml] & readonly;
> public function main() {
>     A tupleValue1 = [1, ""];
>     A tupleValue2 = [1, ""];
> }
> ```

Reply:
Ack. 
In that case, shall we modify the `BTypeReferenceType` constructor to not to hardcode the readonly part?
```java
    public BTypeReferenceType(Type referredType, boolean readOnly)) {
    }
```